src/config/rcxml.c

```diff
@@ -1163,63 +1163,93 @@ rcxml_init(void)
 	rc.workspace_config.min_nr_workspaces = 1;
 }
 
-static struct {
-	const char *binding, *action, *attribute, *value;
+static struct key_combos {
+	const char *binding, *action;
+	struct {
+		const char *name, *value;
+	} attributes[3];
 } key_combos[] = {
-	{ "A-Tab", "NextWindow", NULL, NULL },
-	{ "W-Return", "Execute", "command", "alacritty" },
-	{ "A-F3", "Execute", "command", "bemenu-run" },
-	{ "A-F4", "Close", NULL, NULL },
-	{ "W-a", "ToggleMaximize", NULL, NULL },
-	{ "A-Left", "MoveToEdge", "direction", "left" },
-	{ "A-Right", "MoveToEdge", "direction", "right" },
-	{ "A-Up", "MoveToEdge", "direction", "up" },
-	{ "A-Down", "MoveToEdge", "direction", "down" },
-	{ "W-Left", "SnapToEdge", "direction", "left" },
-	{ "W-Right", "SnapToEdge", "direction", "right" },
-	{ "W-Up", "SnapToEdge", "direction", "up" },
-	{ "W-Down", "SnapToEdge", "direction", "down" },
-	{ "A-Space", "ShowMenu", "menu", "client-menu"},
-	{ "XF86_AudioLowerVolume", "Execute", "command", "amixer sset Master 5%-" },
-	{ "XF86_AudioRaiseVolume", "Execute", "command", "amixer sset Master 5%+" },
-	{ "XF86_AudioMute", "Execute", "command", "amixer sset Master toggle" },
-	{ "XF86_MonBrightnessUp", "Execute", "command", "brightnessctl set +10%" },
-	{ "XF86_MonBrightnessDown", "Execute", "command", "brightnessctl set 10%-" },
-	{ NULL, NULL, NULL, NULL },
+	{ "A-Tab", "NextWindow",
+		{ { NULL, NULL } } },
+	{ "W-Return", "Execute",
+		{ { "command", "alacritty" } } },
```

Comment:
Personally I think this is slightly easier to read:
```c
	{ "A-Tab", "NextWindow", {
		{ NULL, NULL } }
	}, {  "W-Return", "Execute", {
		{ "command", "alacritty" } }
	}, {
		..
```

Also, my C foo is a bit lacking here, do we actually need the `{ NULL, NULL }` ones within the initializer? Usually everything not given should be set to a `0` or `NULL` equivalent but I am not sure if that is also happening for arrays within a struct.

Reply:
I adopted your formatting. I tried various combinations of omitting empty values. GCC seems to require at least one field in the first struct of the array, so I was able to use `{ NULL }` instead of `{ NULL, NULL}`; clang complained about that and required at least one full struct initialization in the array before it was happy. The initialization as implemented seems to be the minimum necessary to keep the compilers happy.